Implement `tsh exec` command, mainly for use in (planned) tsh aliases. (#12629)
gravitational/teleport: docs/pages/setup/reference/cli.mdx

$ tsh mfa rm <device-name>
```

### tsh exec

> Run provided command, by default using a shell, with Teleport-specific environment

Reviewer: I think the first paragraph needs to make a more general statement about why `tsh exec` exists and when it is used. The rest of this reference entry mentions the use of a shell and the Teleport-specific environment variables, so I don't think we need to mention them here. 

If using the provided shell and adding the Teleport-specific environment variables _are_ the purposes of the `tsh exec` command, though, I think we can clarify that a bit more.

To start, what's the difference between using:

```bash
$ tsh exec --shell bash -- -c 'echo "${TSH} ${TSH_COMMAND}"'
```
And:

```BASH
$ export TSH=foo; export TSH_COMMAND=bar; bash -c 'echo "${TSH} ${TSH_COMMAND}"'

Author: A large part of motivation for `tsh exec` are tsh aliases, which are not done yet. 

Outside of that, there is nothing **strictly** new that wasn't possible before: you can write your own scripts, run `tsh env` to get the variables, run `where tsh` to find path to `tsh` binary etc. Yet `tsh exec` automates all of that, which is a convenience.

> variables set. Can be used to create combined commands.

Reviewer: I think this should expand on "combined commands." What are these commands combined with?

Author: The motivating example is in RFD:

```
aliases:
  "connect": "exec -- -c '$TSH login $0 && $TSH ssh $1'"
```

Here we use `exec` to combine `login` and `ssh`.

> 

Reviewer: This should include the command format, e.g., `tsh exec [--shell] -- <command>`. Other command entries in this reference add this after the first paragraph.

The command can be used to run external executables, defaulting to a shell. To
find a usable shell, `tsh exec` will try in order: `$SHELL` env variable, `sh`

> , `bash`. The flag `--shell` allows for overriding of the executable to run,

Reviewer: To have `tsh` execute the command without a shell, would you set `--shell` to an empty string?

Author: `tsh exec` is for executing outside programs. By default, those programs are shells (`$SHELL`, `bash`, `sh`), but you can override the program to run using `--shell` flag. If you set `--shell` to empty string it willf fail to run anything because there is no "" executable.

The flag `--shell` is a bit of a misnomer, but I couldn't find a better name so I left it as is. I'd appreciate suggestions though.

while all arguments passed to command are fed as-is.

> The command will be executed under modified environment, enriched with the following variables:

Reviewer: ```suggestion
The command will be executed under a modified environment, enriched with the following variables:
```

| Name          | Value                                                |
|---------------|------------------------------------------------------|

> | `TSH_COMMAND` | The program being executed.                          |

Reviewer: Is this the full path to the executable? Does it include the arguments?

Author: Just executable, without arguments.

| `TSH`         | The path of the `tsh` binary that invoked the alias. |

Additionally, any variables reported by `tsh env` will also be included. If
the `tsh env` reports no variables (e.g. because the user is not logged in), no

> variables will be added. Currently, the list of variables from `tsh env` may

Reviewer: Is this the full list of variables? If so, instead of "may include," we should say some thing like, "The following environment variables are passed to the command being executed."

include: `TELEPORT_PROXY`, `TELEPORT_CLUSTER`, `TELEPORT_KUBE_CLUSTER`
, `KUBECONFIG`.

#### Examples

Print some environment variables as well as reversed arguments:

```code
$ tsh exec -- -c 'echo $TSH $TSH_COMMAND $2 $1 $0' arg0 arg1 arg2

/bin/tsh /bin/bash arg2 arg1 arg0
```

> Login to cluster `leaf` and ssh as `ubuntu` to `node-1`:

Reviewer: ```suggestion
Log in to cluster `leaf` and SSH as `ubuntu` to `node-1`:
```

```code
$ tsh exec -- -c '$TSH login --proxy=tele.example.com $0 && $TSH ssh $1' leaf ubuntu@node-1
```

> Execute Python script using `python3` as interpreter:

Reviewer: ```suggestion
Execute a Python script using `python3` as the interpreter:
```
